src/test/java/de/tum/in/www1/artemis/metis/MessageIntegrationTest.java

```diff
@@ -386,7 +387,7 @@ void testDecreaseUnreadMessageCountAfterMessageRead() throws Exception {
                 .getConversationParticipants().stream()
                 .filter(conversationParticipant -> !Objects.equals(conversationParticipant.getUser().getId(), postToSave1.getAuthor().getId())).findAny().orElseThrow()
                 .getUnreadMessagesCount();
-
+        await().until(() -> unreadMessages == 0L);
```

Comment:
Good idea would be to add timeout here, so that test will fail in case the read/unread mechanism is broken. Otherwise I would run indefinitely

Reply:
await() has an integrated timeout of 10 seconds so this is not a problem here.